tests/util/pool_tests.rs

```diff
@@ -97,3 +98,33 @@ pub fn test_filesystem_rename(paths: &[&Path]) {
 
     engine.teardown().unwrap();
 }
+
+/// Verify that destroy_filesystems actually deallocates the space
+/// from the thinpool, by that attempting to reinstantiate it using
+/// the same thin id and verifying that it fails.
+pub fn test_thinpool_thindev_destroy(paths: &[&Path]) -> () {
+    let (mut pool, _) = StratPool::initialize("stratis_test_pool",
+                                              &DM::new().unwrap(),
+                                              paths,
+                                              Redundancy::NONE,
+                                              true)
+            .unwrap();
+    let &(_, fs_uuid) = pool.create_filesystems(&["stratis_test_filesystem"])
+        .unwrap()
+        .first()
+        .unwrap();
+
+    let fs_id = pool.get_strat_filesystem(&fs_uuid).unwrap().thin_id();
+
+    pool.destroy_filesystems(&[&fs_uuid]).unwrap();
+
+    // Try to setup a thindev that has been destroyed
+    let dm = DM::new().unwrap();
+    let thindev = ThinDev::setup("stratis_test_thin_dev",
+                                 &dm,
+                                 pool.thinpooldev(),
+                                 fs_id,
+                                 Sectors(128u64));
+    assert!(thindev.is_err());
```

Comment:
Would be good to figure out how to check if the filesystem entry has been removed from the mdv.